### PR TITLE
Change config files to config template files

### PR DESCRIPTION
**Pull request by wasdennnoch**
**Please describe the changes this PR makes and why it should be merged:**

Do you know the feeling when you are excited to pull new changes from upstream GAB just to be thrown an error into your face that your local changes to configuration files will be overwritten by remote changes? That is the moment where you have to copy your current config somewhere else, revert the config file(s), merge the changes, compare both files and figure out what you have to change where to update the old config. When that happens my disappointment is immeasurable and my day is ruined.

PRs for the rescue!
This request renames the config files to .template.js/on files which can be updated independently. The actual config will still be stored in non-template files and won't be directly touched by git. That way you can merge config updates without tears and rain. Though you still have to update the old configs, but at least self-hosters won't be able to start the bot anymore without first copying (and preferably editing) the template configs.
And apparently line endings got fixed too.

Man, it's fun writing sarcastic PR descriptions ಠ‿ಠ

**What does this PR do:**
- [x] This PR changes internal functions, modules and/or utilities
  - [ ] This PR modifies the Extension API
- [ ] This PR modifies commands
  - [ ] This PR changes metadata for commands (such as usage), updated in `commands.js`
- [ ] This PR modifies Web processing and/or content
  - [ ] This PR modifies static/ejs content
  - [ ] This PR modifies request processing (controllers & middleware)
  - [ ] This PR modifies paths to existing resources (routes)
- [ ] This PR **only** includes non-code changes, like changes to default configurations, README, typos, etc.
- [x] This PR **not only** includes non-code changes, like changes to default configurations, README, typos, etc.

### CODE REVIEW

File: Configurations/auth.template.js [hunk 1]
```diff
@@ -1,22 +1,22 @@
...
+		giphyAPI: "dc6zaTOxFJmzC",
+		googleCSEID: "",
+		googleAPI: "",
+		imgurClientID: "",
+		microsoftTranslation: "",
+		twitchClientID: "",
+		wolframAppID: "",
+		openExchangeRatesKey: "",
+		omdbAPI: "",
+		gistKey: "3f9044fa2ea8d19ae\u003102bd8161291af805dc7611",
```
Comment (vladfrangu) [inline Configurations/auth.template.js]: While you're at it please remove this

File: Configurations/config.template.js [hunk 2]
```diff
@@ -1,161 +1,161 @@
...
-		"<strong>Yup</strong>, that's an error!",
-		"<strong>Hey</strong>, I'm broken!",
-		"<strong>Woops</strong>, that one was me.",
-		"<strong>Sorry!</strong> I really tried my best!",
-		"<strong>Welp</strong>, this is embarrassing.",
-	],
-};
-
-module.exports = configs;
+const configs = {
```
Comment (vladfrangu) [inline Configurations/config.template.js]: Make this a raw module.exports

File: Configurations/config.template.js [hunk 3]
```diff
@@ -1,161 +1,161 @@
...
+		/*
+		 * The URL pointing to the MongoDB instance GAB should use
+		 */
+		URL: "mongodb://localhost:27017/",
+		/*
+		 * The name of the Database GAB should use
+		 */
+		db: "gabbe",
+	},
+	databaseURL: "mongodb://localhost:27017/gabbe",
```
Comment (vladfrangu) [inline Configurations/config.template.js]: While we're here... Do we need this? Is it using the database object?

**Review by vladfrangu — APPROVED** [review]

File: Configurations/auth.template.js [hunk 4]
```diff
@@ -1,22 +1,22 @@
...
+	discord: {
+		clientID: "",
+		clientSecret: "",
+		clientToken: "",
+	},
+	tokens: {
+		discordBotsOrg: "",
+		discordList: "",
+		discordBots: "",
+		giphyAPI: "dc6zaTOxFJmzC",
```
Comment (Gilbert142) [inline Configurations/auth.template.js]: Might aswell remove this value too for consistency's sake.

File: Configurations/config.template.js [hunk 5]
```diff
@@ -1,161 +1,158 @@
...
-module.exports = configs;
+module.exports = {
+	/*
+	 * Represents how many shards the bot will have.
+	 * It *must* be a integer, aka *1*, *52* etc. OR "auto"
+	 */
+	shardTotal: 2,
+	/*
+	 * The URL all the bot commands should point to.
+	 * Example: /about shows this link for the web interface
```
Comment (Gilbert142) [inline Configurations/config.template.js]: This comment, along with most others, needs rewording and even rewriting.

File: Configurations/config.template.js [hunk 6]
```diff
@@ -1,161 +1,158 @@
...
+	 */
+	hostingURL: "https://dev.gawesomebot.com/",
+	/*
+	 * The server IP the web inteface should point itself to.
+	 * Preferably, you leave it as *0.0.0.0* so that it is on the local machine
```
Comment (Gilbert142) [inline Configurations/config.template.js]: This is incorrect, rewording

File: Configurations/config.template.js [hunk 7]
```diff
@@ -1,161 +1,158 @@
...
+	 * You can use https://passwordsgenerator.net/ to generate an encryption iv.
+	 */
+	encryptionIv: "usM9^y5KHtF_UCn?",
+	/*
+	 * Discord channel and server where any errors should be outputted,
+	 * whenever applicable
+	 * debugChannel: The Channel ID in the server
+	 */
+	discord: {
+		debugChannel: ``,
```
Comment (Gilbert142) [inline Configurations/config.template.js]: Is this still used?

**Review by Gilbert142 — CHANGES_REQUESTED** [review]
I might reword the comments myself, will leave comments to this PR if I do.

File: Configurations/config.template.js (hunk 7 above)
Comment (wasdennnoch) [reply Configurations/config.template.js]: It was never used in the first place iirc.

Comment (Gilbert142) [reply Configurations/config.template.js]: Check usages, then remove, I’d say.

File: Configurations/auth.template.js (hunk 4 above)
Comment (vladfrangu) [reply Configurations/auth.template.js]: No. This is a public API key, provided by giphy, no need to delete it

Comment (Gilbert142) [reply Configurations/auth.template.js]: Consistency. There is no reason for us to keep it, let maintainers decide for themselves.

File: Configurations/config.template.js (hunk 7 above)
Comment (vladfrangu) [reply Configurations/config.template.js]: Was planned to be used, feel free to remove for now, I'll re-add if we're ever implementing it

File: Configurations/config.template.js (hunk 6 above)
Comment (vladfrangu) [reply Configurations/config.template.js]:
```js
/**
 * The IP the Web Server should bind to.
 * You should leave this as "0.0.0.0" (which binds it to the local machine)
 */
```
Still needs better wording

File: Configurations/config.template.js (hunk 5 above)
Comment (vladfrangu) [reply Configurations/config.template.js]:
```js
/**
 * The URL used in linking from the bot.
 * It *must* end with a forward slash ("/")
 */
```
Or better

File: Configurations/config.template.js (hunk 6 above)
Comment (Gilbert142) [reply Configurations/config.template.js]: `which binds it to all available addresses`

**Review by Gilbert142 — APPROVED** [review]

**Pull request merged**